### PR TITLE
[docs] Document new scroll actions (closes #6072)

### LINKED ISSUE #6072: [docs] Document t.scroll
document #5965

### PR DESCRIPTION
**Pull request by DIRECTcut**
fixes #6072 

Documentation for methods (click to check out how it is going to look): 

[t.scroll](https://directcut.github.io/testcafe/documentation/reference/test-api/testcontroller/scroll.html)
[t.scrollBy](https://directcut.github.io/testcafe/documentation/reference/test-api/testcontroller/scrollby.html)
[t.scrollintoView](https://directcut.github.io/testcafe/documentation/reference/test-api/testcontroller/scrollintoview.html)

### CODE REVIEW

**Review by AlexKamaev — APPROVED** [review]

File: docs/articles/templates/actions/offset-options.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Offset options supply offset parameters for `t.scroll`, `t.scrollBy` and `t.scrollIntoView` actions.
```
Comment (titerman) [inline docs/articles/templates/actions/offset-options.md:1]: how can these options 'supply' parameters?
```suggestion
Offset options apply to `t.scroll`, `t.scrollBy` and `t.scrollIntoView` actions.
```

File: docs/articles/templates/actions/offset-options.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```js
+{
+    offsetX: Number,
+    offsetY: Number,
+}
+```
+
+Parameter                      | Type    | Description                                                                                                                                                 | Default
+------------------------------ | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------
+`offsetX`, `offsetY`           | Number  | Mouse pointer coordinates that define a point where the action is performed or started. If positive, TestCafe calculates coordinates relative to the top-left corner of the target element. If negative, they are calculated relative to the bottom-right corner. | The center of the target element.
```
Comment (titerman) [inline docs/articles/templates/actions/offset-options.md:12]: very very convoluted imo. I'm not sure that even I understand this 100%. Why does it work like this? What decides if the offset applies to the action's starting point or the mouse pointer's target position? You can break it down into smaller chunks if necessary...

**Review by titerman — CHANGES_REQUESTED** [review]

File: docs/articles/documentation/guides/basic-guides/interact-with-the-page.md [hunk 3]
```diff
@@ -408,20 +408,14 @@ This test introduces a `RequestLogger` that logs requests to a location and rece
 
 ### Scroll an Element into View
 
-Since TestCafe scrolls to reach items that are on the page but not on-screen, the TestCafe API does not have a dedicated scroll action.
+Since TestCafe scrolls to reach items that are on the page but not on-screen, there is usually no need to manually scroll the page.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/documentation/guides/basic-guides/interact-with-the-page.md:411]: > As such, very few test scenarios require explicit scroll directives.

Too wordy.

```suggestion
When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. There is usually no need to use the `scroll` action.
```

File: docs/articles/documentation/guides/basic-guides/interact-with-the-page.md [hunk 4]
```diff
@@ -408,20 +408,14 @@ This test introduces a `RequestLogger` that logs requests to a location and rece
...
 
 You can use any action (for example, [hover](#hover)) to scroll towards the desired part of the page.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/documentation/guides/basic-guides/interact-with-the-page.md:413]: `Reveal` doesn't work for me in this context.

```suggestion
Interact with a page element (for example, [hover](#hover)) to scroll it into view.
```

File: docs/articles/documentation/guides/basic-guides/interact-with-the-page.md [hunk 5]
```diff
@@ -408,20 +408,14 @@ This test introduces a `RequestLogger` that logs requests to a location and rece
 
 ### Scroll an Element into View
 
-Since TestCafe scrolls to reach items that are on the page but not on-screen, the TestCafe API does not have a dedicated scroll action.
+Since TestCafe scrolls to reach items that are on the page but not on-screen, there is usually no need to manually scroll the page.
 
 You can use any action (for example, [hover](#hover)) to scroll towards the desired part of the page.  
 
-If you specifically need to scroll the page without any action, use a [ClientFunction](obtain-client-side-info.md).
+If you specifically need to scroll the page without any action, use one of the following methods:
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/documentation/guides/basic-guides/interact-with-the-page.md:415]: `Disrupt` doesn't work here imo.

```suggestion
To scroll the page without interacting with an element, use one of the following methods:
```

File: docs/articles/templates/actions/manual-scroll-warning.md [hunk 6]
```diff
@@ -0,0 +1 @@
+> Important! When you perform actions, TestCafe scrolls the webpage as needed to reach elements. There is usually no need to scroll the page manually.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/templates/actions/manual-scroll-warning.md:1]:
```suggestion
> Important! When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. There is usually no need to use the `scroll` action explicitly.
```

File: docs/articles/documentation/reference/test-api/testcontroller/scroll.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+Parameter   | Type/Value                                        | Description
+----------- | ------------------------------------------------- | --------------------
+`target`&#160;*(optional)* | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the webpage element to scroll. If unspecified, TestCafe scrolls the document body. See [Select Target Elements](#select-target-elements).
+`position`&#160;           | 'left' &#124; 'right' &#124; 'bottom' &#124; 'topLeft' &#124; 'topRight' &#124; 'bottomLeft' &#124; 'bottomRight' &#124; 'center'| TestCafe scrolls the element to this position.
+`options`&#160;*(optional)*| Object | A set of options with additional parameters for the action. See [Options](#options).
+
+## Example
+
+In the example below, TestCafe scrolls the target container to its bottom right.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/testcontroller/scroll.md:41]: Again, `reveal` doesn't work in this context. According to[ MDN's `scroll` definition](https://developer.mozilla.org/en-US/docs/Web/API/Element/scroll),

```
You can scroll the element to a particular set of coordinates inside a given element.
```

```suggestion
In the example below, TestCafe scrolls the target element until the bottom right of that element is inside the viewport.
```

File: docs/articles/documentation/guides/basic-guides/interact-with-the-page.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (titerman) [reply docs/articles/documentation/guides/basic-guides/interact-with-the-page.md:411]: Sounds good to me, except I'd throw out the word 'explicit', it is unnecessary in this particular case.
```suggestion
When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. There is usually no need to use the `scroll` action.
```

File: docs/articles/documentation/guides/basic-guides/interact-with-the-page.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (titerman) [reply docs/articles/documentation/guides/basic-guides/interact-with-the-page.md:413]: lgtm

File: docs/articles/templates/actions/manual-scroll-warning.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (titerman) [reply docs/articles/templates/actions/manual-scroll-warning.md:1]:
```suggestion
> Important! When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. There is usually no need to use the `scroll` action.
```

File: docs/articles/templates/actions/offset-options.md (hunk 2 above)
Comment (DIRECTcut) [reply docs/articles/templates/actions/offset-options.md:12]:
```suggestion
`offsetX`, `offsetY`           | Number  | A point inside of the target element. Before TestCafe performs a scroll, it reveals this point of the element. If positive, TestCafe calculates coordinates relative to the top-left corner of the target element. If negative, they are calculated relative to the bottom-right corner.  | Top left of the target element.
```

File: docs/articles/documentation/reference/test-api/testcontroller/scroll.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+```text
+t.scroll([target,] posX, posY[, options]) → this | Promise<any>
+```
+
+Parameter   | Type/Value                                        | Description
+----------- | ------------------------------------------------- | --------------------
+`target`&#160;*(optional)* | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the webpage element to scroll. If unspecified, TestCafe scrolls the document body. See [Select Target Elements](#select-target-elements).
+`posX`&#160;               | Number | TestCafe scrolls the element to this position on the horizontal axis.
+`posY`&#160;               | Number | TestCafe scrolls the element to this position on the vertical axis.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply docs/articles/documentation/reference/test-api/testcontroller/scroll.md:24]: I think this may be the best we've got

File: docs/articles/templates/actions/offset-options.md (hunk 2 above)
Comment (titerman) [reply docs/articles/templates/actions/offset-options.md:12]: good enough but what do you think about this version?

```suggestion
`offsetX`, `offsetY`           | Number  | Coordinates that define the action's starting point. If positive, TestCafe calculates coordinates relative to the top-left corner of the target element. If negative, they are calculated relative to the bottom-right corner.  | Top left corner of the target element
```

**Review by titerman — APPROVED** [review]

File: docs/articles/documentation/guides/basic-guides/interact-with-the-page.md [hunk 9]
```diff
@@ -408,20 +408,14 @@ This test introduces a `RequestLogger` that logs requests to a location and rece
 
 ### Scroll an Element into View
 
-Since TestCafe scrolls to reach items that are on the page but not on-screen, the TestCafe API does not have a dedicated scroll action.
+When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. There is usually no need to use the `scroll` action.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/guides/basic-guides/interact-with-the-page.md:411]:
```suggestion
When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. You do not need to use the `scroll` action.
```

File: docs/articles/documentation/guides/concepts/typescript-and-coffeescript.md [hunk 10]
```diff
@@ -68,7 +68,7 @@ test('Scroll element into view', async t => {
 });
 ```
 
-To avoid compilation errors, pick element-specific data types, such as [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement)).
+To avoid compilation errors, pick element-specific data types, such as [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement).
 
 The example code below calls the [t.eval](../../reference/test-api/testcontroller/eval.md) method to determine if an ordered list is [reversed](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). The generic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties) does not provide access to the element's `reversed` property. To avoid compilation errors, it is necessary to convert the `list` object to the [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties) data type.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/guides/concepts/typescript-and-coffeescript.md:73]:
```suggestion
The example code below calls the [t.eval](../../reference/test-api/testcontroller/eval.md) method to determine if an ordered list is [reversed](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). The generic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties) does not provide access to the element's `reversed` property. To avoid compilation errors, convert the `list` object to the [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties) data type.
```

File: docs/articles/documentation/reference/test-api/testcontroller/scroll.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
+---
+layout: docs
+title: t.scroll Method
+permalink: /documentation/reference/test-api/testcontroller/scroll.html
+---
+# t.scroll Method
+
+{% include actions/manual-scroll-warning.md %}
+
+Scrolls the `target` element to a specified position. If no `target` is specified, scrolls the document body. Can be chained with other `TestController` methods.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/reference/test-api/testcontroller/scroll.md:10]:
```suggestion
Scrolls the `target` element to a specified position. If no `target` is specified, the document body is scrolled. Can be chained with other `TestController` methods.
```

File: docs/articles/documentation/reference/test-api/testcontroller/scroll.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+Parameter   | Type/Value                                        | Description
+----------- | ------------------------------------------------- | --------------------
+`target`&#160;*(optional)* | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the webpage element to scroll. If unspecified, TestCafe scrolls the document body. See [Select Target Elements](#select-target-elements).
+`position`&#160;           | 'left' &#124; 'right' &#124; 'bottom' &#124; 'topLeft' &#124; 'topRight' &#124; 'bottomLeft' &#124; 'bottomRight' &#124; 'center'| TestCafe scrolls the element to this position.
+`options`&#160;*(optional)*| Object | A set of options with additional parameters for the action. See [Options](#options).
+
+## Example
+
+In the example below, TestCafe reveals the target container's bottom right corner.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/reference/test-api/testcontroller/scroll.md:41]: Why does it 'reveal' the corner? Is it hidden?

File: docs/articles/documentation/reference/test-api/testcontroller/scrollby.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
+---
+layout: docs
+title: t.scrollBy Method
+permalink: /documentation/reference/test-api/testcontroller/scrollby.html
+---
+# t.scrollBy Method
+
+{% include actions/manual-scroll-warning.md %}
+
+Scrolls the `target` element by the given amount of pixels. If no `target` is specified, scrolls the document body. Can be chained with other `TestController` methods.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/reference/test-api/testcontroller/scrollby.md:10]:
```suggestion
Scrolls the `target` element by the specified number of pixels. If no `target` is specified, the document body is scrolled. Can be chained with other `TestController` methods.
```

File: docs/articles/documentation/reference/test-api/testcontroller/scrollintoview.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
+---
+layout: docs
+title: t.scrollIntoView Method
+permalink: /documentation/reference/test-api/testcontroller/scrollintoview.html
+---
+# t.scrollIntoView Method
+
+{% include actions/manual-scroll-warning.md %}
+
+Scrolls parent containers of the `target` element until the element is inside of the viewport. Can be chained with other `TestController` methods.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/reference/test-api/testcontroller/scrollintoview.md:10]:
```suggestion
Scrolls parent containers of the `target` element until the element is in the viewport. Can be chained with other `TestController` methods.
```

File: docs/articles/documentation/reference/test-api/testcontroller/scrollintoview.md [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+```text
+t.scrollIntoView(target [, options]) → this | Promise<any>
+```
+
+Parameter   | Type                                              | Description
+----------- | ------------------------------------------------- | --------------------
+`target`&#160; | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the webpage element to scroll to. See [Select Target Elements](#select-target-elements).
+`options`&#160;*(optional)* | Object | A set of options with additional parameters for the action. See [Options](#options).
+
+The following example shows how to use the `t.scrollIntoView` action to scroll the webpage until the element is in the viewport.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/reference/test-api/testcontroller/scrollintoview.md:21]:
```suggestion
The following example shows how to use the `t.scrollIntoView` action to scroll the webpage until the element is in the viewport:
```

File: docs/articles/templates/actions/manual-scroll-warning.md [hunk 16]
```diff
@@ -0,0 +1 @@
+> Important! When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. There is usually no need to use the `scroll` action.
```
Comment (dirk-pieterse) [inline docs/articles/templates/actions/manual-scroll-warning.md:1]:
```suggestion
> Important! When TestCafe interacts with an off-screen DOM element, it scrolls that element into view. You do not need to use the `scroll` action.
```

**Review by dirk-pieterse — APPROVED** [review]

File: docs/articles/documentation/reference/test-api/testcontroller/scroll.md (hunk 12 above)
Comment (DIRECTcut) [inline docs/articles/documentation/reference/test-api/testcontroller/scroll.md:41]:
```suggestion
In the example below, TestCafe scrolls the target container's bottom right corner into view.
```
